examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+async fn initialize_sdk_from_file(device_path: &PathBuf) -> Result<IronOxide> {
+    if device_path.is_file() {
+        let device_context_file = File::open(&device_path)?;
+        let device_context: DeviceContext = serde_json::from_reader(device_context_file)?;
+        println!("Found DeviceContext in \"{}\"", device_path.display());
+        Ok(ironoxide::initialize(&device_context, &Default::default()).await?)
+    } else {
+        Err(AppErr(
+            format!(
+                "Couldn't open file {} containing DeviceContext",
+                device_path.display()
+            )
+            .to_string(),
+        ))
+    }
+}
+
+// Add the program start time to the end of the specified string to
+// form a unique id string, then create a GroupId from it
+// start-snippet{createGroupId}
+fn create_group_id(id_str: &str) -> Result<GroupId> {
+    let stime = &GLOBAL_STATE.read().unwrap().start_time;
+    let gid = id_str.to_owned() + stime;
+    Ok(GroupId::try_from(gid)?)
+}
+// end-snippet{createGroupId}
+
+// Create a group with the specified ID and name, assuming the current user
+// should be a member and an admin
+// start-snippet{createGroup}
+async fn create_group(
+    sdk: &IronOxide,
+    group_id: &GroupId,
+    name: &str,
+) -> Result<GroupCreateResult> {
+    let opts = GroupCreateOpts::new(
+        Some(group_id.to_owned()),                   // ID
+        Some(GroupName::try_from(name.to_owned())?), // name
+        true,                                        // add as admin
+        true,                                        // add as user
+        None,                                        // owner - defaults to caller
+        vec![],                                      // additional admins
+        vec![],                                      // additional users
+        false,                                       // needs rotation
+    );
+    let group = sdk.group_create(&opts).await?;
+    Ok(group)
+}
+// end-snippet{createGroup}
+
+// Given a customer's data, encrypt the PII and create a Customer struct
+// containing the encrypted data and DEKs
+async fn encrypt_customer(
+    sdk: &IronOxide,
+    id: u32,
+    name: &str,
+    email: &str,
+    group_id: &GroupId,
+) -> Result<Customer> {
+    let encrypt_opts = DocumentEncryptOpts::with_explicit_grants(
+        None,                  // document ID
+        None,                  // document name
+        true,                  // encrypt to self
+        vec![group_id.into()], // users and groups to which to grant access
+    );
+    let enc_result = sdk
+        .document_encrypt_unmanaged(name.as_bytes(), &encrypt_opts)
+        .await?;
+    let enc_name = base64::encode(enc_result.encrypted_data());
+    let enc_name_keys = base64::encode(enc_result.encrypted_deks());
+
+    Ok(Customer {
+        id,
+        name: enc_name,
+        name_keys: enc_name_keys,
+        email: email.to_string(),
+        email_keys: "".to_string(),
```

Comment:
Maybe this is the same question as before, but why aren't you encrypting the email?